.github/workflows/build.yml

```diff
@@ -0,0 +1,155 @@
+      id-token: write # for docker/login to login to NGINX registry
+    runs-on: ${{ github.event_name != 'pull_request' && contains(inputs.image, 'plus') && 'kic-plus' || 'ubuntu-22.04' }}
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+    steps:
+        - name: Checkout Repository
+          uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+          with:
+            ref: ${{ inputs.tag != '' && format('refs/tags/v{0}', inputs.tag) || github.ref }}
+
+        - name: Fetch Cached Artifacts
+          uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
+          with:
+            path: ${{ github.workspace }}/dist
+            key: nginx-gateway-fabric-${{ github.run_id }}-${{ github.run_number }}
+
+        - name: Docker Buildx
+          uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
+          with:
+            driver-opts: network=host
+
+        - name: Setup QEMU
+          uses: docker/setup-qemu-action@68827325e0b33c7199eb31dd4e31fbe9023e06e3 # v3.0.0
+          with:
+            platforms: arm64
+
+        - name: Login to GitHub Container Registry
+          uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+          if: ${{ github.event_name != 'pull_request' && ! contains(inputs.image, 'plus') }}
+          with:
+            registry: ghcr.io
+            username: ${{ github.repository_owner }}
+            password: ${{ secrets.GITHUB_TOKEN }}
+
+        - name: Get Id Token
+          uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+          id: idtoken
+          with:
+            script: |
+              let id_token = await core.getIDToken()
+              core.setOutput('id_token', id_token)
+          if: ${{ github.event_name != 'pull_request' && contains(inputs.image, 'plus')}}
+
+        - name: Login to NGINX Registry
+          uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+          with:
+              registry: docker-mgmt.nginx.com
+              username: ${{ steps.idtoken.outputs.id_token }}
+              password: ${{ github.actor }}
+          if: ${{ github.event_name != 'pull_request' && contains(inputs.image, 'plus') }}
+
+        - name: Docker meta
+          id: meta
+          uses: docker/metadata-action@dbef88086f6cef02e264edb7dbf63250c17cef6c # v5.5.0
+          with:
+            images: |
+              name=ghcr.io/nginxinc/nginx-gateway-fabric,enable=${{ inputs.image == 'ngf' && github.event_name != 'pull_request' }}
+              name=ghcr.io/nginxinc/nginx-gateway-fabric/nginx,enable=${{ inputs.image == 'nginx' && github.event_name != 'pull_request' }}
+              name=docker-mgmt.nginx.com/nginx-gateway-fabric/nginx-plus,enable=${{ inputs.image == 'nginx-plus' && github.event_name != 'pull_request' }}
```

Comment:
are we skipping building if not on a pull request?

Reply:
this is just the tag for the image, we always build `localhost:5000/nginx-gateway-fabric/${{ inputs.image }}` in PRs and depending on the input image, we enable other tags

Reply:
Also, the plus image should be named `nginxplus` for consistency, right?

Reply:
What do you mean for consistency? This was the name that was suggested when we talked about the private registry...but I'm happy to change it

Reply:
Oh didn't realize that. It's just that the default name of the images when building locally is `nginxplus`. I'm not married to either name, but they should probably be the same.

Reply:
Oh I see...in the Makefile 😄 
In NIC all the images have dashes, that's why I defaulted to that one, but I'm not married to it either.

Reply:
@pleshakov @kate-osborn Do you care if we update to `nginx-plus`?

Reply:
Nope, that works 👍

Reply:
👍